vertical-pod-autoscaler/pkg/recommender/Makefile

```diff
@@ -1,33 +1,59 @@
 all: build
 
-TAG?=v0.0.1
-REGISTRY?=k8s.gcr.io
+TAG?=dev
+REGISTRY?=staging-k8s.gcr.io
 FLAGS=
 ENVVAR=
 GOOS?=linux
+COMPONENT=recommender
+FULL_COMPONENT=vpa-${COMPONENT}
 
 deps:
 	go get github.com/tools/godep
 
 build: clean deps
 	$(ENVVAR) GOOS=$(GOOS) godep go build ./...
-	$(ENVVAR) GOOS=$(GOOS) godep go build -o recommender
+	$(ENVVAR) GOOS=$(GOOS) godep go build -o ${COMPONENT}
+
+build-binary: clean deps
+	$(ENVVAR) GOOS=$(GOOS) godep go build -o ${COMPONENT}
 
 test-unit: clean deps build
 	$(ENVVAR) godep go test --test.short -race ./... $(FLAGS)
 
-docker:
+docker-build:
+ifndef REGISTRY
+	ERR = $(error REGISTRY is undefined)
+	$(ERR)
+endif
+ifndef TAG
+	ERR = $(error TAG is undefined)
+	$(ERR)
+endif
+	docker build --pull -t ${REGISTRY}/${FULL_COMPONENT}:${TAG} .
+
+docker-push:
 ifndef REGISTRY
 	ERR = $(error REGISTRY is undefined)
 	$(ERR)
 endif
-	docker build --pull -t ${REGISTRY}/vpa-recommender:${TAG} .
-	docker push ${REGISTRY}/vpa-recommender:${TAG}
+ifndef TAG
+	ERR = $(error TAG is undefined)
+	$(ERR)
+endif
+	docker push ${REGISTRY}/${FULL_COMPONENT}:${TAG}
+
+docker-builder:
+	docker build -t autoscaling-builder ../../../builder
```

Comment:
Using the same builder for CA and VPA will mean we always have to do things like upgrade Go version simultaneously etc. Do we want to do this?